lib/steep/type_inference/logic_type_interpreter.rb

```diff
@@ -408,6 +416,63 @@ def evaluate_method_call(env:, type:, receiver:, arguments:)
         end
       end
 
+      def evaluate_union_method_call(node:, env:, receiver:, receiver_type:)
+        call_type = typing.call_of(node: node) rescue nil
+        return unless call_type.is_a?(Steep::TypeInference::MethodCall::Typed)
+
+        truthy_types = [] #: Array[AST::Types::t]
+        falsy_types = [] #: Array[AST::Types::t]
+
+        receiver_type.types.each do |type|
+          case type
+          when AST::Types::Name::Base
+            ancestors = factory.definition_builder.ancestor_builder.singleton_ancestors(type.name).ancestors
+            ancestors.find do |ancestor|
+              call_type.method_decls.find do |decls|
+                if ancestor.name == decls.method_def.defined_in
+                  return_type = factory.type(decls.method_def.type.type.return_type)
+                  truthy, falsey = factory.partition_union(return_type)
+                  truthy_types << type if truthy
+                  falsy_types << type if falsey
+                  true
+                end
+              end
+            end
+          when AST::Types::Nil, AST::Types::Literal
+            ancestors = factory.definition_builder.ancestor_builder.singleton_ancestors(type.back_type.name).ancestors
+            ancestors.find do |ancestor|
+              call_type.method_decls.find do |decls|
+                if ancestor.name == decls.method_def.defined_in
+                  return_type = factory.type(decls.method_def.type.type.return_type)
+                  truthy, falsey = factory.partition_union(return_type)
+                  truthy_types << type if truthy
+                  falsy_types << type if falsey
+                  true
+                end
+              end
+            end
+          else
+            truthy_types << type
+            falsy_types << type
+          end
```

Comment:
I handled only `Name::*,` `Nil`, and `Literal.` But there are kinds of AST::Types. How do we evaluate them?